src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -59,7 +59,13 @@ public override void VisitField(IFieldSymbol symbol)
                 AddPunctuation(SyntaxKind.DotToken);
             }
 
-            if (symbol.ContainingType.TypeKind == TypeKind.Enum)
+            if (symbol.AssociatedSymbol is IPropertySymbol associatedProperty)
```

Comment:
>if (symbol.AssociatedSymbol is IPropertySymbol associatedProperty) [](http://example.com/codeflow?start=12&length=66)

Is this check going to work for symbols imported from metadata? Please add a test.